tests/Doctrine/Tests/DBAL/Functional/ExceptionTest.php

```diff
@@ -314,7 +315,8 @@ public function testConnectionExceptionSqLite($mode, $exceptionClass)
     public function getSqLiteOpenConnection()
     {
         return array(
-            array(0000, '\Doctrine\DBAL\Exception\ConnectionException'),
+            // mode 0 is considered read-only on Windows
+            array(0000, '\\' !== DIRECTORY_SEPARATOR ? '\Doctrine\DBAL\Exception\ConnectionException' : '\Doctrine\DBAL\Exception\ReadOnlyException'),
```

Comment:
I'd rather not infer the host-OS based on the directory-separator. Maybe look at the `PHP_OS` constant instead? I've seen other projects check the presence of the `PHP_WINDOWS_VERSION_BUILD` constant as well.